ext/src/zpages/tracez_data_aggregator.cc

```diff
@@ -0,0 +1,110 @@
+#include "opentelemetry/ext/zpages/tracez_data_aggregator.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext
+{
+namespace zpages
+{
+
+TracezDataAggregator::TracezDataAggregator(std::shared_ptr<TracezSpanProcessor> span_processor)
+{
+  tracez_span_processor_ = span_processor;
+}
+
+const std::map<std::string, std::unique_ptr<AggregatedInformation>>& TracezDataAggregator:: GetAggregatedData()
+{
+  AggregateSpans();
+  return aggregated_data_;
+}
+
+LatencyBoundaryName TracezDataAggregator::GetLatencyBoundary(opentelemetry::sdk::trace::Recordable* recordable)
+{
+  auto recordable_duration = recordable->GetDuration();
+  for(unsigned int boundary = 0; boundary < kLatencyBoundaries.size()-1; boundary++)
+  {
+    if(kLatencyBoundaries[boundary] <= recordable_duration && recordable_duration < kLatencyBoundaries[boundary+1])return (LatencyBoundaryName)boundary;
```

Comment:
Nit: a possible optimization will be to compare only with the upper limit, because the lower limit is the upper limit of the previous bucket.